File: captainhcg/go/404-sum-of-left-leaves.go
var ret int
func sumOfLeftLeaves(root *TreeNode) int {
    ret = 0
    var helper func(node *TreeNode, isLeft bool)
    helper = func(node *TreeNode, isLeft bool) {
        if node.Left == nil && node.Right == nil{
            if isLeft{
                ret += node.Val
            }
            return
        }
        if node.Left != nil{
            helper(node.Left, true)
        }
        if node.Right != nil{
            helper(node.Right, false)
        }
    }
    if root != nil {
        helper(root, false)
    }
    return ret
}

File: bwu/binary_tree/404-sum-of-left-leaves.py
class Solution(object):
    def sumOfLeftLeaves(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        sum = 0
        if not root:
            return 0
            
        if root.left and not root.left.left and not root.left.right:
            sum += root.left.val
            
        sum += self.sumOfLeftLeaves(root.left) + self.sumOfLeftLeaves(root.right)
        return sum

    def sumOfLeftLeaves2(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        res = 0
        if not root:
            return res
            
        stack = [root]
        while stack:
            node = stack.pop(0)
            if node.left:
                if not node.left.left and not node.left.right:
                    res += node.left.val
                else:
                    stack.append(node.left)
            if node.right:
                if node.right.left or node.right.right:
                    stack.append(node.right)
                    
        return res
